google/cloud/bigquery/client.py

```diff
@@ -2456,6 +2457,7 @@ def load_table_from_dataframe(
         project: str = None,
         job_config: LoadJobConfig = None,
         parquet_compression: str = "snappy",
+        parquet_use_compliant_nested_type: bool = True,
```

Comment:
I'm more than happy to not expose this!! I agree that it's just plain wrong.

I was just thinking about anyone who needs a work around for a table they've already created with a 'wrong' format.

Reply:
I think I should clarity this. The reason it might break old users tables is that the format that `pyarrow` uses for nested lists by default is `list<item: list<item: value>>`. By default (without `use_compliant_nested_type`) it turns this into `list<list<item: value>>` when writing to Parquet. 

When `enable_list_inference` is off in BigQuery this ends up as a `ARRAY<STRUCT> in BigQuery where the `STRUCT` looks like:
```
{
    "list: {
        "item: value
    }
}
```

When `enable_list_inference` is on in BigQuery this ends up as a `ARRAY<STRUCT> in BigQuery where the `STRUCT` looks like:
```
{
    "item": value
}
```

When PyArrow is told to `use_compliant_nested_type` it outputs `list<list<element: value>>`.

When `enable_list_inference` is off in BigQuery this ends up as a `ARRAY<STRUCT> in BigQuery where the `STRUCT` looks like:
```
{
    "list: {
        "element: value
    }
}
```

When `enable_list_inference` is on in BigQuery this ends up as a `ARRAY<value> in BigQuery. Which is what it should be!!!!

My point being that if we switch on `use_compliant_nested_type` without giving the option to turn it off the `item` becomes `element` always. This will be guaranteed to be incompatible with schemas created with the legacy version.

Does that make sense?

Reply:
If you're happy to have this breaking change but 'officially' support list types for the pandas connector - then I'm happy to not expose it! And will make the change?

Reply:
Yes, this change seems to be what's needed to officially support list types. 😁

Reply:
Makes sense. I hadn't really expected anyone to actually use that rather strange schema...

Since https://github.com/googleapis/python-bigquery/issues/19 is still open, I don't consider lists supported as of today, so I'm okay making this breaking change.

Reply:
That said, if you think it would be disruptive we can consider adding a field now, but then we immediately remove it in `google-cloud-bigquery` 3.0 🤔

Reply:
I think it's reasonable to make the breaking change. If someone cares enough to maintain the old behaviour they can write the Dataframe using whatever options they want manually and then use whatever JobConfig they like with `load_table_from_file`. Doesn't seem worth it to add the option and immediately remove it. Especially, as as you say... the old schema is.... difficult to use.

Reply:
Cool. Let's remove it, then. Especially since folks have a workaround if they really need the unsupported schema.